cvc5/Solver.lean

```diff
@@ -406,82 +697,198 @@ def mkInteger (tm : TermManager) : Int → Term :=
 
 See `cvc5.Kind` for a description of the parameters.
 
-- `kind` The kind of the operator.
-- `args` The arguments (indices) of the operator.
+- `kind`: The kind of the operator.
+- `args`: The arguments (indices) of the operator.
 
 If `args` is empty, the `Op` simply wraps the `cvc5.Kind`. The `Kind` can be used in `Solver.mkTerm`
 directly without creating an `Op` first.
 -/
 @[extern "termManager_mkOpOfIndices"]
-opaque mkOpOfIndices : TermManager → (kind : Kind) → (args : Array Nat) → Op
+opaque mkOpOfIndices : TermManager → (kind : Kind) → (args : Array Nat) → Except Error Op
+
+@[inherit_doc mkOpOfIndices]
+def mkOpOfIndices! tm kind args :=
+  mkOpOfIndices tm kind args
+  |> Error.unwrap!
 
 /-- Create operator of kind:
 
 - `Kind.DIVISIBLE` (to support arbitrary precision integers)
 
 See `cvc5.Kind` for a description of the parameters.
 
-- `kind` The kind of the operator.
-- `arg` The string argument to this operator.
+- `kind`: The kind of the operator.
+- `arg`: The string argument to this operator.
 
 -/
 @[extern "termManager_mkOpOfString"]
-opaque mkOpOfString : TermManager → (kind : Kind) → (arg : String) → Op
+private opaque mkOpOfString : TermManager → (kind : Kind) → (arg : String) → Except Error Op
+
+/-- Create divisibility-by operator, supports arbitrary precision. -/
+def mkOpDivisible (tm : TermManager) (n : Nat) (_valid : 0 < n := by simp) : Op :=
+  tm.mkOpOfString Kind.DIVISIBLE (toString n)
+  |> Error.unwrap!
 
 /-- Create n-ary term of given kind.
 
-- `kind` The kind of the term.
-- `children` The children of the term.
+- `kind`: The kind of the term.
+- `children`: The children of the term.
 -/
 @[extern "termManager_mkTerm"]
-opaque mkTerm : TermManager → (kind : Kind) → (children : Array Term := #[]) → Term
+opaque mkTerm : TermManager → (kind : Kind) → (children : Array Term := #[]) → Except Error Term
+
+@[inherit_doc mkTerm]
+def mkTerm! tm kind children :=
+  mkTerm tm kind children
+  |> Error.unwrap!
 
 /-- Create n-ary term of given kind from a given operator.
 
 Create operators with `mkOp`.
 
-- `op` The operator.
-- `children` The children of the term.
+- `op`: The operator.
+- `children`: The children of the term.
 -/
 @[extern "termManager_mkTermOfOp"]
-opaque mkTermOfOp : TermManager → (op : Op) → (children : Array Term := #[]) → Term
+opaque mkTermOfOp : TermManager → (op : Op) → (children : Array Term := #[]) → Except Error Term
+
+@[inherit_doc mkTermOfOp]
+def mkTermOfOp! tm op children :=
+  mkTermOfOp tm op children
+  |> Error.unwrap!

```

Comment:
Use macro. Avoid use of `|>` if the number of consecutive parentheses is less than 3: `(mkTermOfOp tm op children).unwrap!`